md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/AnalyticsAppAPI.java

```diff
@@ -0,0 +1,281 @@
+            }
+            else if (string.startsWith("appproperties_ddpurl")) {
+                jpaProperties = Dao2TableUtil.buildJPAProperties("analytics-app", "ddp-url", map.get(string), "DDP url");
+                propertiesList.add(jpaProperties);
+                LOGGER.debug("appproperties_ddpurl" + map.get(string));
+            }
+
+
+
+
+        }
+
+
+        com.wipro.ats.bdre.md.dao.jpa.Process parentProcess = Dao2TableUtil.buildJPAProcess(37, processName, processDescription, 1,busDomainId);
+        Users users=new Users();
+        users.setUsername(principal.getName());
+        parentProcess.setUsers(users);
+        parentProcess.setUserRoles(userRolesDAO.minUserRoleId(principal.getName()));
+        com.wipro.ats.bdre.md.dao.jpa.Process childProcess1 = Dao2TableUtil.buildJPAProcess(38, "subprocess1 of "+processName, "subprocess1 of "+processDescription, 1,busDomainId);
+        //    com.wipro.ats.bdre.md.dao.jpa.Process childProcess2 = Dao2TableUtil.buildJPAProcess(38, "subprocess2 of "+processName, "subprocess1 of "+processDescription, 1,busDomainId);
+
+        List<Process> childProcesses = new ArrayList<Process>();
+        childProcesses.add(childProcess1);
+        //     childProcesses.add(childProcess2);
+
+
+        List<com.wipro.ats.bdre.md.dao.jpa.Process> processList = processDAO.createAnalyticsAppJob(parentProcess, childProcesses, propertiesList);
+        List<com.wipro.ats.bdre.md.beans.table.Process> tableProcessList = Dao2TableUtil.jpaList2TableProcessList(processList);
+        Integer counter = tableProcessList.size();
+        for (com.wipro.ats.bdre.md.beans.table.Process process:tableProcessList) {
+            process.setCounter(counter);
+            process.setTableAddTS(DateConverter.dateToString(process.getAddTS()));
+            process.setTableEditTS(DateConverter.dateToString(process.getEditTS()));
+        }
+        restWrapper = new RestWrapper(tableProcessList, RestWrapper.OK);
+        LOGGER.info("Process and Properties for Analytics App process inserted by" + principal.getName());
+
+
+        return restWrapper;
+    }
+
+    @RequestMapping(value = {"/industries"}, method = RequestMethod.GET)
+    @ResponseBody
+    public RestWrapper list(Principal principal) {
+
+        RestWrapper restWrapper = null;
+        try {
+
+            List<AnalyticsApps> analyticsAppsList = new ArrayList<AnalyticsApps>();
+            List<String> industriesList = analyticsAppsDAO.listIndustries();
+            LOGGER.info(industriesList);
+            for (String industry : industriesList) {
+                AnalyticsApps analyticsApp = new AnalyticsApps();
+                analyticsApp.setIndustryName(industry);
+                analyticsAppsList.add(analyticsApp);
+            }
+
+            restWrapper = new RestWrapper(analyticsAppsList, RestWrapper.OK);
+            LOGGER.info("All records listed from Properties by User:" + principal.getName());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "All records listed from Properties by User:" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1192)